runtime/script_loader.cpp

```diff
@@ -237,7 +267,18 @@ bool ScriptLoader::load_top_level_script(const char *path, MutableHandleValue re
   JS::NonIncrementalGC(cx, JS::GCOptions::Shrink, JS::GCReason::API);
 
   // Execute the top-level module script.
-  return MODULE_MODE
-    ? ModuleEvaluate(cx, module, result)
-    : JS_ExecuteScript(cx, script, result);
+  if (!MODULE_MODE) {
+    return JS_ExecuteScript(cx, script, result);
+  }
+  
+  if (!ModuleEvaluate(cx, module, result)) {
+    return false;
+  }
+  
+  // modules return the top-level await promise in the result value
+  // we don't currently support TLA, instead we reassign result
+  // with the module namespace
```

Comment:
Would you mind filing an issue for TLA? It really shouldn't be hard to support, so we should perhaps just do it before too long. (But of course not as a prerequisite to or part of this PR.)